nox/tasks.py

```diff
@@ -199,28 +205,33 @@ def filter_manifest(
         # (which is an error condition later).
         manifest.filter_by_keywords(global_config.keywords)
 
-    # Return the modified manifest.
-    return manifest
+    if not manifest and not global_config.list_sessions:
+        logger.error("No sessions selected after filtering by keyword.")
+        return 3
 
+    # If the global_config is set to an empty list, and a list was not
+    # requested, and no filtering was done,
+    if (
+        global_config.sessions is not None
+        and not global_config.sessions
+        and not global_config.list_sessions
+        and not global_config.keywords
+        and not global_config.pythons
+    ):
+        manifest.filter_by_name(global_config.sessions)
+        print("No sessions selected. Please select a session with -s <session name>.\n")
```

Comment:
Should we use the logger here? It would make the message stand out from the session listing.

```suggestion
        logger.warning(
            "No sessions selected. Please select a session with -s <session name>.\n"
        )
```

We'd need to adapt the new tests to use `caplog.text` instead of `capsys.readouterr().out`.

Reply:
Is it an error / warning though? The session listing is a print.